silkworm/rpc/core/evm_trace.cpp

```diff
@@ -747,6 +781,34 @@ void VmTraceTracer::on_pre_check_failed(const evmc_result& /*result*/, const evm
     vm_trace_.code = "0x" + silkworm::to_hex(ByteView{msg.input_data, msg.input_size});
 }
 
+void VmTraceTracer::fill_call_gas_info(TraceOp& trace_op, const evmone::ExecutionState& execution_state, const intx::uint256* stack_top, const int stack_height, const silkworm::IntraBlockState& intra_block_state) {
+    auto op_code = trace_op.op_code;
+    if (op_code == evmc_opcode::OP_CALL || op_code == evmc_opcode::OP_CALLCODE || op_code == evmc_opcode::OP_STATICCALL || op_code == evmc_opcode::OP_DELEGATECALL || op_code == evmc_opcode::OP_CREATE || op_code == evmc_opcode::OP_CREATE2) {
+        fix_call_gas_info_ = std::make_unique<FixCallGasInfo>(FixCallGasInfo{execution_state.msg->depth, 0, metrics_[op_code].gas_cost, trace_op});
+        const auto value = stack_top[-2];  // value
+        if (value != 0) {
+            fix_call_gas_info_->gas_cost += 9000;
+        }
+        if (op_code == OP_CALL) {
+            if (op_code == OP_CALL && stack_height >= 7 && value != 0) {
+                fix_call_gas_info_->stipend = 2300;  // for CALLs with value, include stipend
+            }
+            const auto call_gas = stack_top[0];                              // gas
+            const auto dst = intx::be::trunc<evmc::address>(stack_top[-1]);  // dst
+
+            if ((value != 0 || execution_state.rev < EVMC_SPURIOUS_DRAGON) && !intra_block_state.exists(dst)) {
+                fix_call_gas_info_->gas_cost += 25000;
+            }
+            SILK_DEBUG << "DebugTracer::evaluate_call_fixes:"
+                       << " call_gas: " << call_gas
+                       << " dst: " << dst
+                       << " value: " << value
+                       << " gas_cost: " << fix_call_gas_info_->gas_cost
+                       << " stipend: " << fix_call_gas_info_->stipend;
+        }
+    }
+}
```

Comment:
Some more comments with references to YP or Geth/Erigon implementation would be very helpful here.

Reply:
Maybe better to evmone implementation